microcosm_pubsub/caching/memcached.py

```diff
@@ -0,0 +1,88 @@
+"""
+Serialization helpers for caching.
+
+"""
+from enum import IntEnum, unique
+from json import dumps, loads
+
+from pymemcache.client.base import Client
+from pymemcache.test.utils import MockMemcacheClient
+
+from microcosm_pubsub.caching.base import CacheBase
+
+
+@unique
+class SerializationFlag(IntEnum):
+    """
+    Used by caching backends to control how to serialize
+    or deserialize cached values.
+
+    Memcached is the primary use case.
+
+    """
+    STRING = 1
+    JSON = 2
+
+
+def json_serializer(key, value):
+    """
+    Simple JSON serializer for use with caching backends
+    that only support string/bytes value storage.
+
+    Memcached is the primary use case.
+
+    """
+    if isinstance(value, str) or isinstance(value, bytes):
```

Comment:
apparently this memcached driver does some conversion to bytes under the hood for strings, so including both here